server/src/main/java/com/defold/extender/ExtenderUtil.java

```diff
@@ -368,15 +389,153 @@ static private String stripMergeKey(String name)
         return name;
     }
 
+    // Merges b on top of a:
+    //  a = {"key": valueA, "list" : ListA}
+    //  b = {"key": valueB, "list" : ListB}
+    //  -> {"key": valueB, "list" : merge(ListA, ListB)}
+    static public Map<String, Object> mergeMaps(Map<String, Object> a, Map<String, Object> b) throws ExtenderException {
+        Map<String, Object> context = new HashMap<>(a);
+
+        // Clean the names of the previous context
+        Set<String> originalKeys = new HashSet<>(context.keySet());
+        for (String k : originalKeys) {
+            boolean isMergeOp = isMergeOp(k);
+            String key = stripMergeKey(k);
+            if (!isMergeOp)
+            {
+                Object v = context.get(k);
+                context.remove(k);
+                context.put(key, v);
+            }
+        }
+
+        Set<String> keys = b.keySet();
+        for (String k : keys) {
+
+            boolean isMergeOp = isMergeOp(k); // true by default
+            String key = stripMergeKey(k);
+
+            Object v1 = context.getOrDefault(key, context.getOrDefault(k, null));
+            Object v2 = b.get(k);
+
+            if (v1 == null && v2 == null) {
+                // Simply skip keys that hold no values at all
+                context.remove(key);
+                continue;
+            }
+
+            if (v1 != null && v2 != null && !v1.getClass().equals(v2.getClass())) {
+                throw new ExtenderException(String.format("Wrong manifest context variable type for %s: Expected %s, got %s: %s", k, v1.getClass().toString(), v2.getClass().toString(), v2.toString()));
+            }
+            if (v2 != null && v2 instanceof List && !ExtenderUtil.isListOfStrings((List<Object>) v2)) {
+                throw new ExtenderException(String.format("The context variables only support lists of strings. Got %s (type %s)", v2.toString(), v2.getClass().getCanonicalName()));
+            }
+
+            if (v1 != null && v2 != null && v1 instanceof List) {
+                if (isMergeOp) {
+                    v1 = ExtenderUtil.mergeLists((List<String>) v1, (List<String>) v2);
+                } else {
+                    v1 = v2;
+                }
+                context.put(key, v1);
+            }
+            else {
+                context.put(key, v2 != null ? v2 : v1);
+            }
+        }
+        return context;
+    }
+
+    // Merges two object
+    //  * Same class
+    //  * Uses the public fields
+    //  * Merges Lists and Maps
+    //  * Maps can override values by using the "_replace" name on variables
+    // Merges other on top of dst:
+    //  dst = {"key": valueA, "list" : ListA, "keyA": valueA}
+    //  other = {"key": valueB, "list" : ListB, , "keyB": valueB}
+    //  -> dst: {"key": valueB, "list" : merge(ListA, ListB), , "keyA": valueA, , "keyB": valueB}
+    //@SuppressWarnings("unchecked")
+    public static <T extends Object> void mergeObjects(T dst, T other) throws ExtenderException {
```

Comment:
Helper function to merge two instances of the same type.
E.g. we use it to merge properties of two "PlatformConfig"